packages/devtools_app/lib/src/memory/flutter/memory_controller.dart

```diff
@@ -597,35 +1017,130 @@ class MemoryTimeline {
   List<Map> processMemoryLogFileData() {
     assert(controller.offline);
     assert(offlineData.isNotEmpty);
-    return _processData(offlineData, 0);
+    return _processData(startingIndex);
   }
 
   List<Map> processLiveData([bool reloadAllData = false]) {
     assert(!controller.offline);
     assert(liveData.isNotEmpty);
 
-    final usedSize = chartData.used.length;
-    if (usedSize != liveData.length || reloadAllData) {
-      return _processData(liveData, usedSize);
+// *****************************************
+// TODO(terry): Handle engineChartData too.
+// *****************************************
+
+    if (endingIndex - startingIndex != liveData.length || reloadAllData) {
+      // Process the data received (startingDataIndex is the last sample).
+      final args = _processData(endingIndex < 0 ? 0 : endingIndex);
+
+      // Debugging data - to enable remove logical not operator.
+      if (!true) {
+        final DateFormat mFormat = DateFormat('hh:mm:ss.mmm');
+        final startDT = mFormat.format(DateTime.fromMillisecondsSinceEpoch(
+            liveData[startingIndex].timestamp.toInt()));
+        final endDT = mFormat.format(DateTime.fromMillisecondsSinceEpoch(
+            liveData[endingIndex].timestamp.toInt()));
+        print('Time range Live data start: $startDT, end: $endDT');
+      }
+
+      return args;
     }
 
     return [];
   }
 
+  List<Map> recomputeOfflineData(int displayInterval) {
+    assert(displayInterval > 0);
+
+    // Compute a new starting index from length - N minutes.
+    final timeLastSample = data.last.timestamp;
+    var dataIndex = data.length - 1;
+    for (; dataIndex > 0; dataIndex--) {
```

Comment:
ditto. this looks like the same code used elsewhere - would benefit from a helper

Reply:
Again function removed.